pkg/server/nodemanager/node_controller.go

```diff
@@ -246,10 +256,27 @@ func (c *NodeController) startNodes(nodes []*api.Node, image cloud.Image) {
 			klog.Errorf("Error creating node in registry: %v", err)
 			continue
 		}
+		image := getImageForInstance(newNode.Spec.InstanceType, images)
 		go c.startSingleNode(newNode, image, metadata)
 	}
 }
 
+func getImageForInstance(instType string, images []cloud.Image) cloud.Image {
+	var image cloud.Image
+	for _, img := range images {
+		isMacInst := strings.HasPrefix(instType, "mac")
+		isMacImg := strings.Contains(img.Name, "elotl-kipmac-")
+		isDefaultImg := strings.Contains(img.Name, "elotl-kip-")
+		if isMacInst && isMacImg {
+			image = img
+		}
+		if !isMacInst && isDefaultImg {
+			image = img
+		}
+	}
+	return image
```

Comment:
We could return an “empty” image here if images aren’t named properly.

Either return an error, or a boolean indicating whether an image has been found in the list to avoid weird bugs in the future.